consensus/polybft/polybft.go

```diff
@@ -397,31 +412,21 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 			}
 
 			// initialize EIP1559Burn SC
-			if config.Params.BurnContract != nil &&
-				len(config.Params.BurnContract) == 1 &&
-				!polyBFTConfig.NativeTokenConfig.IsMintable {
-				var contractAddress types.Address
-				for _, address := range config.Params.BurnContract {
-					contractAddress = address
+			if burnContractAddress, ok := getBurnContractAddress(config, polyBFTConfig); ok {
```

Comment:
Can we use here already gotten `burnContractAddress` and `ok` variables, that we got on line 155?